Source/WebCore/page/scrolling/ScrollingTree.cpp

```diff
@@ -309,17 +318,40 @@ void ScrollingTree::setOverlayScrollbarsEnabled(bool enabled)
     m_overlayScrollbarsEnabled = enabled;
 }
 
-bool ScrollingTree::commitTreeState(std::unique_ptr<ScrollingStateTree>&& scrollingStateTree)
+void ScrollingTree::removeNode(ScrollingNodeID nodeID)
 {
-    SetForScope inCommitTreeState(m_inCommitTreeState, true);
-    Locker locker { m_treeLock };
+    m_latchingController.nodeWasRemoved(nodeID);
+    if (auto node = m_nodeMap.take(nodeID)) {
+        auto nodeList = m_nodeMapPerFrame.find(node->frameIdentifier());
+        if (nodeList != m_nodeMapPerFrame.end())
+            nodeList->value.remove(node);
+        node->willBeDestroyed();
+    }
+}
 
+bool ScrollingTree::commitTreeStateInternal(std::unique_ptr<ScrollingStateTree>&& scrollingStateTree, std::optional<LayerHostingContextIdentifier> identifier)
+{
     bool rootStateNodeChanged = scrollingStateTree->hasNewRootStateNode();
 
     LOG(ScrollingTree, "\nScrollingTree %p commitTreeState", this);
```

Comment:
I think much more logging about subframe tree connections would be useful.

Reply:
Right will re-add the logging I had locally.